src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16587,7 +16587,12 @@ def index_equals(self, other: "SnowflakeQueryCompiler") -> bool:
             )
         }
 
-        rows = join_result.result_frame.ordered_dataframe.agg(agg_exprs).collect()
+        try:
+            rows = join_result.result_frame.ordered_dataframe.agg(agg_exprs).collect()
+        except Exception:
```

Comment:
One potential issue we have here it might incorrectly return False for other type of errors se well for example network failures.
Can you please specifically check of exception where this should be False? Another option is to perform this type check before doing value check and return False if types don't align.